src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs

```
@@ -108,21 +112,70 @@ public async Task PushDockerImageToECR(Recommendation recommendation, string rep
+                var targetFramework = recommendation.ProjectDefinition.TargetFramework ?? string.Empty;
+
+                // Elastic Beanstalk doesn't currently have .NET 7 preinstalled.
+                var unavailableFramework = new List<string> { "net7.0" };
+                var frameworkNames = new Dictionary<string, string> { { "net7.0", ".NET 7" } };
+                if (unavailableFramework.Contains(targetFramework))
+                {
+                    _interactiveService.LogInfoMessage($"Using self-contained publish since AWS Elastic Beanstalk does not currently have {frameworkNames[targetFramework]} preinstalled");
+                    recommendation.DeploymentBundle.DotnetPublishSelfContainedBuild = true;
+                    return;
+                }
+
+                var beanstalkPlatformSetting = recommendation.Recipe.OptionSettings.FirstOrDefault(x => x.Id.Equals("ElasticBeanstalkPlatformArn"));
+                if (beanstalkPlatformSetting != null)
+                {
+                    var beanstalkPlatformSettingValue = _optionSettingHandler.GetOptionSettingValue<string>(recommendation, beanstalkPlatformSetting);
+                    var beanstalkPlatformSettingValueSplit = beanstalkPlatformSettingValue?.Split("/");
+                    if (beanstalkPlatformSettingValueSplit?.Length != 3)
+                        throw new InvalidElasticBeanstalkPlatformException(DeployToolErrorCode.InvalidElasticBeanstalkPlatform, $"The selected Elastic Beanstalk platform version '{beanstalkPlatformSettingValue}' is invalid.");
+                    var beanstalkPlatformName = beanstalkPlatformSettingValueSplit[1];
+                    if (!Version.TryParse(beanstalkPlatformSettingValueSplit[2], out var beanstalkPlatformVersion))
+                        throw new InvalidElasticBeanstalkPlatformException(DeployToolErrorCode.InvalidElasticBeanstalkPlatform, $"The selected Elastic Beanstalk platform version '{beanstalkPlatformSettingValue}' is invalid.");
```

Is this format part of Elastic Beanstalk's "contract"? I would hate to penalize users by throwing an exception if EB ever changed the convention of these strings. 

I'm wondering if it'd be safer to just proceed and possibly let the deployment fail, since users could then set `Self Contained Build` themselves.

---

Done